src/NServiceBus.Core/Pipeline/ConnectorContextExtensions.cs

```diff
@@ -218,19 +218,19 @@ public static IOutgoingPhysicalMessageContext CreateOutgoingPhysicalMessageConte
         /// <summary>
         /// Creates a <see cref="IAuditContext" /> based on the current context.
         /// </summary>
-        public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, IIncomingPhysicalMessageContext sourceContext)
+        public static IAuditContext CreateAuditContext(this ForkConnector<IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, TimeSpan? timeToBeReceived, IIncomingPhysicalMessageContext sourceContext)
         {
             Guard.AgainstNull(nameof(sourceContext), sourceContext);
 
             var connector = (IForkConnector<IIncomingPhysicalMessageContext, IIncomingPhysicalMessageContext, IAuditContext>)forkConnector;
-            return connector.CreateAuditContext(message, auditAddress, sourceContext);
+            return connector.CreateAuditContext(message, auditAddress, timeToBeReceived, sourceContext);
         }
 
-        internal static IAuditContext CreateAuditContext(this IForkConnector<IIncomingPhysicalMessageContext, IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, IIncomingPhysicalMessageContext sourceContext)
+        internal static IAuditContext CreateAuditContext(this IForkConnector<IIncomingPhysicalMessageContext, IIncomingPhysicalMessageContext, IAuditContext> forkConnector, OutgoingMessage message, string auditAddress, TimeSpan? timeToBeReceived, IIncomingPhysicalMessageContext sourceContext)
         {
             _ = forkConnector;
```

Comment:
Still don't get why discarding here is necessary. Just so that the tooling doesn't mark it as not being used? If so wouldn't we have to do this then consistently in the other methods too?

Reply:
no idea either, lets handle that separately